### PR TITLE
QOwnNotes

### PR DESCRIPTION
**Pull request by stobbsm**
Adding QOwnNotes to flathub. Worked on getting the appdata file to validate, but it works now.

### CODE REVIEW

**Comment by stobbsm** [comment]
This was done with the express permission of Patrizio Bekerle (pbek), the original author of QOwnNotes.
See Issue: https://github.com/pbek/QOwnNotes/issues/928

File: org.qownnotes.QOwnNotes.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+	"app-id": "org.qownnotes.QOwnNotes",
+	"runtime": "org.kde.Platform",
+	"runtime-version": "5.10",
+	"sdk": "org.kde.Sdk",
+	"command": "QOwnNotes",
+	"rename-desktop-file": "PBE.QOwnNotes.desktop",
+	"rename-icon": "QOwnNotes",
+	"finish-args": [
+		"--socket=x11",
```
Comment (Alexander-Wilms) [inline org.qownnotes.QOwnNotes.json]: If it works under Wayland, you should add the permission

File: org.qownnotes.QOwnNotes.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		"--share=network",
+		"--device=dri",
+		"--filesystem=home"
+	],
+	"modules": [
+		{
+			"name": "QOwnNotes",
+			"buildsystem": "simple",
```
Comment (Alexander-Wilms) [inline org.qownnotes.QOwnNotes.json]: You can use buildsystem: qmake and use a source of type shell to modify release.h

File: org.qownnotes.QOwnNotes.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	"runtime-version": "5.10",
+	"sdk": "org.kde.Sdk",
+	"command": "QOwnNotes",
+	"rename-desktop-file": "PBE.QOwnNotes.desktop",
+	"rename-icon": "QOwnNotes",
+	"finish-args": [
+		"--socket=x11",
+		"--share=network",
+		"--device=dri",
+		"--filesystem=home"
```
Comment (Alexander-Wilms) [inline org.qownnotes.QOwnNotes.json]: It should work without the filesystem permission due to xdg-desktop-portal

File: org.qownnotes.QOwnNotes.json (hunk 2 above)
Comment (Alexander-Wilms) [reply org.qownnotes.QOwnNotes.json]: I think you can leave the subdir as is. Ideally, the make install step should install the appdata as well, but in the meantime, you can add it as another shell command source

File: org.qownnotes.QOwnNotes.json (hunk 1 above)
Comment (Alexander-Wilms) [reply org.qownnotes.QOwnNotes.json]: Also, the performance under X is better if you add a share=ipc permission.

**Comment by stobbsm** [comment]
Updated to fix the issues. Thanks for the help and guidance @Alexander-Wilms

**Comment by Alexander-Wilms** [comment]
No problem. You need to add this as well:

```
{
"type": "shell",
"commands": ["install -Dm644 ../org.qownnotes.QOwnNotes.appdata.xml /app/share/metainfo/org.qownnotes.QOwnNotes.appdata.xml"]
}
```

**Comment by stobbsm** [comment]
Actually, it seems to find it without needing the shell script. Probably because of the common name. Worked perfectly on my fedora 28 install.

**Comment by stobbsm** [comment]
Pushed a change to restore the home directory share. Without it, the application has to be setup on first run everytime.

**Comment by Alexander-Wilms** [comment]
You can try if persist=~/.config or whatever directory it needs access to is enough.

**Comment by stobbsm** [comment]
It stores notes in the ~/Nextcloud/Notes folder, or wherever is specified. I tried persisting them, but it didn't work as expected, still asking to set everything up every launch. There is good reason that it needs home file system access.

**Comment by stobbsm** [comment]
I'll talk with pbek about changing the license. We are going to work out a way to automatically update the appdata.xml and manifest file (if needed) when he does releases, so hopefully this won't be a problem from here on out.

**Comment by nedrichards** [comment]
Yeah, that'd be cool. The licence isn't a blocker, and if it's upstream it's obviously fine. Currently the appdata isn't being put in the correct place/building correctly, which is causing the build to fail: https://flathub.org/builds/#/builders/2/builds/2285

**Comment by stobbsm** [comment]
How do I fix that issue? Where is the appdata.xml supposed to go to be picked up properly? I thought it was /app/share/appdata

**Comment by nedrichards** [comment]
I'm going to look at it myself locally but this usually happens because something isn't automatically linkable (e.g. the appdata doesn't refer to the desktop file correctly or there isn't a launchable element at some point) and appstream-glib wants to pick up combine information from the appdata, desktop file and icons in order to create the final appstream information.

**Comment by nedrichards** [comment]
bot, build org.qownnotes.QOwnNotes

**Comment by nedrichards** [comment]
short answer - it wasn't actually being installed. I split it out into a different module so you can easily drop it when there's an upstream release that includes and correctly installs the appdata. Once this builds fine, it all looks good to me.

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/org.qownnotes.QOwnNotes

**Pull request closed without merging**